nemoguardrails/llm/taskmanager.py

```diff
@@ -109,12 +109,14 @@ def _render_string(
         template_str: str,
         context: Optional[dict] = None,
         events: Optional[List[dict]] = None,
+        out_variables: Optional[dict] = None,
     ) -> str:
         """Render a template using the provided context information.
 
         :param template_str: The template to render.
         :param context: The context for rendering the prompt.
         :param events: The history of events so far.
+        :param out_variables: If not None the dict will be populated with variables set in the template
```

Comment:
I might be wrong, but maybe 
```suggestion
        :param out_variables: If not None, the dict will be populated with the variables that were set in the template after rendering. This allows the caller to access the values of the variables used in the rendered template.

```